Source/Core/Core/IPC_HLE/WII_IPC_HLE_Device_usb_bt_base.h

```diff
@@ -0,0 +1,137 @@
+// Copyright 2016 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include "Common/Assert.h"
+#include "Core/Core.h"
+#include "Core/IPC_HLE/WII_IPC_HLE.h"
+#include "Core/IPC_HLE/WII_IPC_HLE_Device.h"
+
+constexpr u16 BT_INFO_SECTION_LENGTH = 0x460;
+
+class CWII_IPC_HLE_Device_usb_oh1_57e_305_base : public IWII_IPC_HLE_Device
+{
+public:
+  CWII_IPC_HLE_Device_usb_oh1_57e_305_base(u32 device_id, const std::string& device_name)
+      : IWII_IPC_HLE_Device(device_id, device_name)
+  {
+  }
+  virtual ~CWII_IPC_HLE_Device_usb_oh1_57e_305_base() override = default;
+
+  virtual IPCCommandResult Open(u32 command_address, u32 mode) override = 0;
+  virtual IPCCommandResult Close(u32 command_address, bool force) override = 0;
+  IPCCommandResult IOCtl(u32 command_address) override
```

Comment:
Please put the implementation of this function, CtrlMessage's constructor, CtrlBuffer's constructor, FillBuffer(), and `CWII_IPC_HLE_Device_usb_oh1_57e_305_stub`'s  Open and DoState functions in the .cpp file (the stub can probably be moved to its own .cpp/.h files).

This way you don't dump assert definitions, logging definitions, core functions, and byte swapping utilities into the header file.